src/Transactions/TransferWithMemo.cs

```diff
@@ -28,17 +49,72 @@ private static byte[] Serialize(CcdAmount amount, AccountAddress receiver, OnCha
     {
         using var memoryStream = new MemoryStream((int)(
             sizeof(TransactionType) +
+            CcdAmount.BytesLength +
             AccountAddress.BytesLength +
-            OnChainData.MaxLength +
-            CcdAmount.BytesLength));
+            OnChainData.MaxLength));
         memoryStream.WriteByte(TransactionType);
         memoryStream.Write(receiver.ToBytes());
         memoryStream.Write(memo.ToBytes());
         memoryStream.Write(amount.ToBytes());
         return memoryStream.ToArray();
     }
 
-    public override ulong GetTransactionSpecificCost() => 300;
+    /// <summary>
+    /// Create a "transfer with memo" payload from a serialized as bytes.
+    /// </summary>
+    /// <param name="bytes">The payload as bytes.</param>
+    /// <param name="output">Where to write the result of the operation.</param>
+    public static bool TryDeserial(byte[] bytes, out (TransferWithMemo?, string? Error) output)
+    {
+        var minSize = sizeof(TransactionType) + AccountAddress.BytesLength + CcdAmount.BytesLength;
+        if (bytes.Length < minSize)
+        {
+            var msg = $"Invalid length in `TransferWithMemo.TryDeserial`. Expected at least {minSize}, found {bytes.Length}";
+            output = (null, msg);
+            return false;
+        };
+        if (bytes[0] != TransactionType)
+        {
+            var msg = $"Invalid transaction type in `Transfer.TryDeserial`. expected {TransactionType}, found {bytes[0]}";
+            output = (null, msg);
+            return false;
+        };
+
+        var trxTypeLength = sizeof(TransactionType);
+        var accountLength = (int)AccountAddress.BytesLength;
+        var amountLength = (int)CcdAmount.BytesLength;
+        var memoLength = bytes.Length - trxTypeLength - accountLength - amountLength;
```

Comment:
**Request change:** This is relying on the provided bytes to only contain exactly the payload bytes, which we don't really require anywhere else. 

The length of the memo is encoded in the payload, using UInt16BE:
See https://github.com/Concordium/concordium-base/blob/main/haskell-src/Concordium/Types.hs#L807.